src/datachain/lib/dc.py

```diff
@@ -331,22 +346,80 @@ def from_storage(
             chain = DataChain.from_storage("s3://my-bucket/my-dir")
             ```
         """
-        func = get_file(type)
-        return (
-            cls(
-                path,
+        file_type = get_file_type(type)
+
+        if client_config is None:
+            client_config = {}
+
+        if anon:
+            client_config["anon"] = True
+
+        session = Session.get(
+            session,
+            catalog=kwargs.get("catalog"),
+            client_config=client_config,
+            in_memory=in_memory,
+        )
+
+        client, path = Client.parse_url(
+            uri, session.catalog.cache, **session.catalog.client_config
+        )
+
+        # clean path without globs
+        lst_path = (
+            posixpath.dirname(path)
+            if glob.has_magic(os.path.basename(os.path.normpath(path)))
+            or client.fs.isfile(uri)
```

Comment:
can we now treat file as a single entry dataset? (and avoid indexing the whole parent bucket) - can be a followup, but seems an important one

Reply:
Currently, If user enters specific file  it's parent folder is listed .. e.g
```python
DataChain.from_storage("s3://my-bucket/animals/dogs/dog5.jpg")
```
Bucket listing will have the whole `dogs` folder and dataset holding data will be called `lst_s3://my-bucket/animals/dogs/`

We can do a followup to optimize this if needed.

Reply:
yes, I understand the logic (it is the same as it was before). Mu question was - if it's easier to drop it now and do it properly right away. By properly I mean - index a single file (may be even don't cache it).

Reply:
It think this is very needed.

Reply:
can be a followup though

Reply:
I would rather do it in followup